code/modules/mob/inventory.dm

```diff
@@ -364,14 +364,16 @@
 	drop_all_held_items()
 
 
-/mob/living/carbon/proc/check_obscured_slots(transparent_protection)
+/mob/living/carbon/proc/check_obscured_slots(transparent_protection, opaque_vulnerability)
 	var/obscured = NONE
 	var/hidden_slots = NONE
 
 	for(var/obj/item/I in get_equipped_items())
 		hidden_slots |= I.flags_inv
 		if(transparent_protection)
 			hidden_slots |= I.transparent_protection
+		if(opaque_vulnerability)
+			hidden_slots &= ~I.opaque_vulnerability
```

Comment:
These two can't seem to be both TRUE together, are they?

I think this is a better as enum constants (or better yet, since I don't really like arguments that create branches, separate procs).